service/src/main/java/org/apache/griffin/core/job/JobController.java

```diff
@@ -115,8 +120,19 @@ public JobHealth getHealthInfo() {
     }
 
     @RequestMapping(value = "/jobs/trigger/{id}", method = RequestMethod.POST)
-    @ResponseStatus(HttpStatus.NO_CONTENT)
-    public void triggerJob(@PathVariable("id") Long id) throws SchedulerException {
-        jobService.triggerJobById(id);
+    @ResponseStatus(HttpStatus.OK)
+    public JobInstanceBean triggerJob(@PathVariable("id") Long id, @RequestBody(required = false) String request) throws SchedulerException {
+        return jobService.triggerJobById(id, extractTimeOut(request));
+    }
+
+    private long extractTimeOut(String request) {
+        long timeout = 0;
```

Comment:
body is marked as optional, but this will return zero timeout if not provided -- does not look like correct behavior

Reply:
Why not? The default timeout value is zero, since we do not need to wait for the creation of a JobInstanceBean. If it is important to return a JobInstanceBean, we can set a timeout in the request.

Reply:
Default behavior without waiting could be very confusing for API users, I'm reluctant to have no-wait behavior by default.

Reply:
@chemikadze on the other hand, such behavior corresponds to existing behavior, if I got it right. Otherwise it may happen that updated API introduces delays on requests where users perhaps don't expect them yet. Of course it would be amended by default value 0 for timeout in config... Well... I feel like this situation about timeout needs a bit more consideration.